From fb8c66ce29002907af8614d6f46b7c6cce28fa05 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppetlabs.com>
Date: Fri, 15 Aug 2014 14:02:23 -0700
Subject: [PATCH] (PUP-3008) Guard constant initialization against re-init.

---
 lib/win32/windows/constants.rb | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/win32/windows/constants.rb b/lib/win32/windows/constants.rb
index b6e753d..95eb665 100644
--- a/lib/win32/windows/constants.rb
+++ b/lib/win32/windows/constants.rb
@@ -117,13 +117,13 @@ module Windows
 
     WAIT_OBJECT_0 = 0
     WAIT_TIMEOUT = 0x00000102
-    INFINITE = 0xFFFFFFFF
+    INFINITE ||= 0xFFFFFFFF
 
     IDLE_CONTROL_CODE = 0
 
     DELETE = 0x00010000
     FORMAT_MESSAGE_FROM_SYSTEM    = 0x00001000
-    FORMAT_MESSAGE_IGNORE_INSERTS = 0x00000200
+    FORMAT_MESSAGE_IGNORE_INSERTS ||= 0x00000200
 
     NO_ERROR = 0
 
@@ -133,11 +133,11 @@ module Windows
 
     # Errors
 
-    ERROR_INSUFFICIENT_BUFFER = 122
+    ERROR_INSUFFICIENT_BUFFER ||= 122
     ERROR_MORE_DATA = 234
     ERROR_FILE_NOT_FOUND = 2
     ERROR_RESOURCE_TYPE_NOT_FOUND = 1813
     ERROR_RESOURCE_NAME_NOT_FOUND = 1814
-    WAIT_FAILED = 0xFFFFFFFF
+    WAIT_FAILED ||= 0xFFFFFFFF
   end
 end
-- 
1.8.4.msysgit.0

