From b7a369c0193e5baf8b11ee57961a52c1d1b49a91 Mon Sep 17 00:00:00 2001
From: Michael Smith <michael.smith@puppetlabs.com>
Date: Fri, 15 Aug 2014 14:00:36 -0700
Subject: [PATCH] (PUP-3008) Guard constant initialization against re-init.

---
 lib/win32/windows/constants.rb | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/lib/win32/windows/constants.rb b/lib/win32/windows/constants.rb
index fbcc118..cf7bca1 100644
--- a/lib/win32/windows/constants.rb
+++ b/lib/win32/windows/constants.rb
@@ -23,13 +23,13 @@ module Windows
     REG_EXPAND_SZ = 2
 
     ERROR_SUCCESS = 0
-    ERROR_INSUFFICIENT_BUFFER = 122
+    ERROR_INSUFFICIENT_BUFFER ||= 122
 
     BUFFER_SIZE = 1024 * 64
     MAX_SIZE    = 512
     MAX_STRINGS = 16
-    INFINITE    = 0xFFFFFFFF
-    WAIT_FAILED = 0xFFFFFFFF
+    INFINITE    ||= 0xFFFFFFFF
+    WAIT_FAILED ||= 0xFFFFFFFF
 
     BASE_KEY = "SYSTEM\\CurrentControlSet\\Services\\EventLog\\"
 
@@ -49,7 +49,7 @@ module Windows
     DONT_RESOLVE_DLL_REFERENCES = 0x00000001
     LOAD_LIBRARY_AS_DATAFILE    = 0x00000002
 
-    FORMAT_MESSAGE_IGNORE_INSERTS = 0x00000200
+    FORMAT_MESSAGE_IGNORE_INSERTS ||= 0x00000200
     FORMAT_MESSAGE_FROM_HMODULE   = 0x00000800
     FORMAT_MESSAGE_ARGUMENT_ARRAY = 0x00002000
   end
-- 
1.8.4.msysgit.0

